router.go

```diff
@@ -63,39 +72,76 @@ func (m MapRouter) Register(rs []transport.Procedure) {
 			panic("Expected procedure name not to be empty string in registration")
 		}
 
+		if r.Encoding != "" {
```

Comment:
Supported encodings is open-ended, so we have to treat a blank encoding as a wild card. We do not have a list of supported encodings to enumerate to fill the holes in the (service, procedure, encoding) table if an encoding is not specified.